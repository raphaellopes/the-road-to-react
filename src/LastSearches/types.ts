export type LastSearchesProps = {
  lastSearches: string[];
  onLastSearch: (searchTerm: string) => void;
}
