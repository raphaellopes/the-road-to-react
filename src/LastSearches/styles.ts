import styled from 'styled-components';

export const Container = styled.div`
  margin-bottom: 16px;
  padding-bottom: 16px;
  border-bottom: 1px solid rgba(0,0,0,0.2);

  > button {
    &:not(:last-child) {
      margin-right: 8px;
    }
  }
`;

export const Title = styled.h3``;
