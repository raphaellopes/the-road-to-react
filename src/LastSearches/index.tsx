import React, { FC } from 'react';

import { ButtonSmall } from '../styles';
import { LastSearchesProps } from './types';
import { Container, Title } from './styles';

const LastSearches:FC<LastSearchesProps> = ({ lastSearches, onLastSearch }) => (
  <Container>
    <Title>Last searches:</Title>
    {lastSearches.map((searchTerm, index) => (
      <ButtonSmall
        key={searchTerm + index}
        type="button"
        onClick={() => onLastSearch(searchTerm)}
      >
        {searchTerm}
      </ButtonSmall>
    ))}
  </Container>
)

export default LastSearches;
