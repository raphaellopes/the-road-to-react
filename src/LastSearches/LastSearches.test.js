import { render, screen, fireEvent, act } from '@testing-library/react';
import LastSearches from './index';

describe('LastSearches', () => {
  const props = {
    lastSearches: ['React', 'Redux'],
    onLastSearch: jest.fn()
  }

  test('renders snapshot', () => {
    const { container } = render(<LastSearches {...props} />)
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the component with its properties', () => {
    render(<LastSearches {...props} />);
    expect(screen.getByText('React')).toBeInTheDocument();
    expect(screen.getByText('Redux')).toBeInTheDocument();
  });

  test('calls onLastSearch with its value', () => {
    render(<LastSearches {...props} />);
    fireEvent.click(screen.getByText('React'));
    expect(props.onLastSearch).toHaveBeenCalledTimes(1);
    expect(props.onLastSearch).toHaveBeenCalledWith('React');
  });
});
