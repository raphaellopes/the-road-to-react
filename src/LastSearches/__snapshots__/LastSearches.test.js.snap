// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastSearches renders snapshot 1`] = `
.c1 {
  background: transparent;
  border: 1px solid #171212;
  padding: 5px;
  cursor: pointer;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c1:hover {
  background: #171212;
  color: #ffffff;
}

.c1:hover > svg > g {
  fill: #ffffff;
  stroke: #ffffff;
}

.c2 {
  padding: 5px;
}

.c0 {
  margin-bottom: 16px;
  padding-bottom: 16px;
  border-bottom: 1px solid rgba(0,0,0,0.2);
}

.c0 > button:not(:last-child) {
  margin-right: 8px;
}

<div
  class="c0"
>
  <h3
    class=""
  >
    Last searches:
  </h3>
  <button
    class="c1 c2"
    type="button"
  >
    React
  </button>
  <button
    class="c1 c2"
    type="button"
  >
    Redux
  </button>
</div>
`;
